library/src/androidTest/java/org/xmtp/android/library/GroupTest.kt

```diff
@@ -0,0 +1,208 @@
+        boClient.conversations.newGroup(listOf(caro.walletAddress.lowercase()))
+        val groups = boClient.conversations.listGroups()
+        assertEquals(groups.size, 2)
+    }
+
+    @Test
+    fun testCanListGroupsAndConversations() {
+        boClient.conversations.newGroup(listOf(alix.walletAddress.lowercase()))
+        boClient.conversations.newGroup(listOf(caro.walletAddress.lowercase()))
+        try {
+            boClient.conversations.newConversation(alix.walletAddress)
+        } catch(e: Exception) {
+            boClient.conversations.newConversation(alix.walletAddress.lowercase())
+        }
+        val convos = boClient.conversations.list(includeGroups = true)
+        assertEquals(convos.size, 3)
+    }
+
+    @Test
+    fun testCanSendMessageToGroup() {
+        val group = boClient.conversations.newGroup(listOf(alix.walletAddress.lowercase()))
+        group.send("howdy")
+        group.send("gm")
+        assertEquals(group.messages().first().body, "gm")
+        assertEquals(group.messages().size, 3)
+
+        val sameGroup = alixClient.conversations.listGroups().last()
+        assertEquals(sameGroup.messages().size, 2)
+        assertEquals(sameGroup.messages().first().body, "gm")
+    }
+
+    @Test
+    fun testCanSendContentTypesToGroup() {
+        Client.register(codec = ReactionCodec())
+
+        val group = boClient.conversations.newGroup(listOf(alix.walletAddress.lowercase()))
+        group.send("gm")
+        val messageToReact = group.messages()[0]
+
+        val reaction = Reaction(
+            reference = messageToReact.id,
+            action = ReactionAction.Added,
+            content = "U+1F603",
+            schema = ReactionSchema.Unicode
+        )
+
+        group.send(content = reaction, options = SendOptions(contentType = ContentTypeReaction))
+
+        val messages = group.messages()
+        assertEquals(messages.size, 3)
+        val content: Reaction? = messages.first().content()
+        assertEquals("U+1F603", content?.content)
+        assertEquals(messageToReact.id, content?.reference)
+        assertEquals(ReactionAction.Added, content?.action)
+        assertEquals(ReactionSchema.Unicode, content?.schema)
+    }
+
+    @Test
+    fun testCanStreamGroupMessages() = kotlinx.coroutines.test.runTest {
+        val group = boClient.conversations.newGroup(listOf(alix.walletAddress.lowercase()))
+
+        group.streamMessages().test {
+            group.send("hi")
+            assertEquals("hi", awaitItem().body)
+            awaitComplete()
+        }
+    }
+
+    @Test
+    fun testCanStreamGroups() = kotlinx.coroutines.test.runTest {
+        boClient.conversations.streamGroups().test {
+            val conversation =
+                boClient.conversations.newGroup(listOf(alix.walletAddress.lowercase()))
+            conversation.send(content = "hi")
+            assertEquals("hi", awaitItem().messages().first().body)
```

Comment:
I'd expect "hi" to be the second item, after the membership change message

Reply:
The messaging list is in descending order by default. So what ever the last message to be sent is the first message.

Reply:
Hmmm. Then I don't know. Are you able to see any messages come into Kotlin-land?

Reply:
The streaming doesn't seem to be working. But I can list messages no problem and sync. I think this is just a streaming issue.